pkg/bitmap/bitmap.go

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bitmap
+
+import (
+	"encoding/binary"
+	"math"
+	"sync"
+
+	lbm "github.com/openacid/low/bitmap"
+)
+
+// BitMap is a struct which provides the Get or Set of bits map.
+type BitMap struct {
+	lock        sync.RWMutex
+	bm          []uint64
+	maxBitIndex int32
```

Comment:
could we just make it a type of `uint32` ?
Then we could ignore validation of it. Otherwise I think we had better validate it when initialization.

Reply:
add the validate of all parameters.

Reply:
done.